common/types.go

```diff
@@ -186,6 +186,42 @@ func (h Hash) Value() (driver.Value, error) {
 	return h[:], nil
 }
 
+// GetLastNumBitsFromHash returns the last num bits of the hash as int64
+func (h Hash) GetLastNumBitsFromHash(num int) int64 {
+	if num > 64 {
+		return 0
+	}
+	termHash := h
+	bits := byteSliceToBits(termHash[:])
+	bitSlice, _ := bitSliceToInt64(bits[len(bits)-num:])
```

Comment:
This will panic if num=0, should add a check and return -1 in that case too.

Reply:
I think panic again is ok

Reply:
ok